client/core/core.go

```diff
@@ -7383,18 +7389,19 @@ func (c *Core) connectDEX(acctInfo *db.AccountInfo, temporary ...bool) (*dexConn
 	dc.WsConn = conn
 	dc.connMaster = dex.NewConnectionMaster(conn)
 
-	// At this point, we have a valid dexConnection object whether or not we can
-	// actually connect. In any return below, we return the dexConnection so it
-	// may be tracked in the c.conns map (and listed as a known DEX). TODO:
-	// split the above code into a dexConnection constructor, and the below into
-	// a startDexConnection function so we don't have the anti-pattern of
-	// returning a non-nil object with a non-nil error and requiring the caller
-	// to check both!
+	return dc, nil
+}
 
+// startDexConnection attempts to connect the provided dexConnection. dc must be
+// a new dexConnection returned from newDEXConnection above. Callers can choose
+// to stop reconnect retries and any current goroutine for the provided
+// dexConnection using dc.connMaster.Disconnect().
+func (c *Core) startDexConnection(acctInfo *db.AccountInfo, dc *dexConnection) error {
 	// Start listening for messages. The listener stops when core shuts down or
 	// the dexConnection's ConnectionMaster is shut down. This goroutine should
 	// be started as long as the reconnect loop is running. It only returns when
 	// the wsConn is stopped.
+	listen := dc.reportingConnects == 1
```

Comment:
Should be `listen := dc.broadcastingConnect()` ?

Reply:
Yeah, that works too.